--- a/package/network/services/dnsmasq/files/dnsmasq.init	2022-06-06 13:23:08.155881200 +0800
+++ b/package/network/services/dnsmasq/files/dnsmasq.init	2022-06-06 13:24:54.839425900 +0800
@@ -931,6 +931,11 @@
 	append_bool "$cfg" noping "--no-ping"
 	append_bool "$cfg" rapidcommit "--dhcp-rapid-commit"
 	append_bool "$cfg" scriptarp "--script-arp"
+	
+	append_bool "$cfg" filter_aaaa "--filter-aaaa"
+	append_bool "$cfg" filter_https "--filter-https"
+	append_bool "$cfg" filter_unknown "--filter-unknown"
+	append_parm "$cfg" mini_ttl "--min-ttl"
 
 	append_parm "$cfg" logfacility "--log-facility"
 	config_get logfacility "$cfg" "logfacility"
