--- a/package/network/services/dnsmasq/files/dnsmasq.init	2022-06-06 19:15:14.082932000 +0800
+++ b/package/network/services/dnsmasq/files/dnsmasq.init	2022-06-06 19:15:53.578399000 +0800
@@ -933,6 +933,7 @@
 	append_bool "$cfg" scriptarp "--script-arp"
 
 	append_bool "$cfg" filter_aaaa "--filter-aaaa"
+	append_parm "$cfg" mini_ttl "--min-ttl"
 
 	append_parm "$cfg" logfacility "--log-facility"
 	config_get logfacility "$cfg" "logfacility"
