--- dnsmasq.init	2022-06-07 19:03:06.430334900 +0800
+++ 2dnsmasq.init	2022-06-07 19:05:07.195971300 +0800
@@ -933,6 +933,9 @@
 	append_bool "$cfg" scriptarp "--script-arp"
 
 	append_bool "$cfg" filter_aaaa "--filter-aaaa"
+	append_bool "$cfg" filter_https "--filter-https"
+	append_bool "$cfg" filter_unknown "--filter-unknown"
+	append_parm "$cfg" mini_ttl "--min-ttl"
 
 	append_parm "$cfg" logfacility "--log-facility"
 	config_get logfacility "$cfg" "logfacility"
