--- mt7620.mk	2022-06-11 13:31:45.511660100 +0800
+++ 2mt7620.mk	2022-06-11 13:34:02.219369300 +0800
@@ -1270,16 +1270,17 @@
   BLOCKSIZE := 128k
   PAGESIZE := 2048
   KERNEL_SIZE := 4096k
+  KERNEL := $(KERNEL_DTB) | uImage lzma
   IMAGE_SIZE := 32768k
   UBINIZE_OPTS := -E 5
   IMAGES += kernel1.bin rootfs0.bin breed-factory.bin factory.bin
   IMAGE/kernel1.bin := append-kernel | check-size $$$$(KERNEL_SIZE)
-  IMAGE/rootfs0.bin := append-ubi | check-size
+  IMAGE/rootfs0.bin := append-ubi | check-size $$$$(IMAGE_SIZE)
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | check-size
   IMAGE/breed-factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | \
 			     append-kernel | pad-to $$(KERNEL_SIZE) | \
-			     append-ubi | check-size
+			     append-ubi | check-size $$$$(IMAGE_SIZE)
   DEVICE_VENDOR := Xiaomi
   DEVICE_MODEL := Mi Router R3
   DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb-ohci uboot-envtools
